core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -70,11 +72,12 @@ public Expression residual() {
 
   @Override
   public Iterable<FileScanTask> split(long splitSize) {
-    if (file.format().isSplittable()) {
-      return () -> new SplitScanTaskIterator(splitSize, this);
-    } else {
-      return ImmutableList.of(this);
+    if (file.splitOffsets() != null) {
```

Comment:
The first check should still be whether the format is splittable. If it isn't, we should not split even if there are recommended locations.